tlatools/org.lamport.tlatools/test/util/AstNode.java

```diff
@@ -284,15 +316,31 @@ public static enum Kind {
 		 */
 		private static final Set<Kind> unused = new HashSet<Kind>();
 		
+		/**
+		 * The subset of nodes that are exclusively used by PlusCal.
+		 */
+		private static final Set<Kind> pcal = new HashSet<Kind>();
```

Comment:
Nit: for `enum` types, [`EnumSet`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/EnumSet.html) is far more memory- and cpu-efficient than `HashSet`. Efficiency isn't really a concern here, but `EnumSet` is still a nice thing to know about. :)

Reply:
Oooh, very nice API!